Foundation/include/Poco/NumberFormatter.h

```diff
@@ -190,17 +272,32 @@ class Foundation_API NumberFormatter
 		/// right justified and zero-padded in a field having at least the
 		/// specified width.
 
-	static std::string formatHex(unsigned long long value, bool prefix = false);
+	[[deprecated("use formatHex with options instead")]]
+	static std::string formatHex(unsigned long long value, bool prefix);
 		/// Formats a 64-bit integer value in hexadecimal notation.
 		/// If prefix is true, "0x" prefix is prepended to the
 		/// resulting string.
 
-	static std::string formatHex(unsigned long long value, int width, bool prefix = false);
+	static std::string formatHex(unsigned long long value, Options options = Options::OPT_NONE);
+		/// Formats a unsigned long long value in hexadecimal notation.
+		/// Options (see NumberFormatter::Options) define the format of the
+		/// generated string.
+
+	[[deprecated("use formatHex with options instead")]]
+	static std::string formatHex(unsigned long long value, int width, bool prefix);
 		/// Formats a 64-bit integer value in hexadecimal notation,
 		/// right justified and zero-padded in a field having at least
 		/// the specified width. If prefix is true, "0x" prefix is
 		/// prepended to the resulting string.
 
+	static std::string formatHex(unsigned long long value, int width, Options options = Options::OPT_NONE);
+		/// Formats a 64-bit integer value in hexadecimal notation,
```

Comment:
In some places we it is documented just that use integer value but in some places we have complete type. Which one it should be?

Reply:
I reused the text that was there before but I agree: it can be improved.

Reply:
Resolved in new commit.